add explanation about component containers. (#4902)
ros2/ros2_documentation: source/Tutorials/Intermediate/Composition.rst

Now that we have seen the basic operation of components, we can discuss a few more advanced topics.

Component container types
^^^^^^^^^^^^^^^^^^^^^^^^^

> As introduced in :doc:`Component container <../../Concepts/Intermediate/About-Composition>`, there is a few component container types with options.

Reviewer: ```suggestion
As introduced in :doc:`Component container <../../Concepts/Intermediate/About-Composition>`, there are a few component container types with different options.
```

Also, can we link directly to the section here?

You can choose the most appropriate component container type for your requirement.

> * ``component_container`` (No options / parameters available)

Reviewer: What about using a numbered list here?

Author: this is not step-by-step procedure, so i think bullet points are just fine.

.. code-block:: bash

>    ros2 run rclcpp_components component_container

Reviewer: Could you indent these codeblocks so that they appear under their respective bullet points?

> * ``component_container_mt`` with ``MultiThreadedExecutor`` composed of 4 threads.

Reviewer: ```suggestion
* ``component_container_mt`` with ``MultiThreadedExecutor`` composed of 4 threads.

  * The ``thread_num`` parameter option is available to specify the number of threads.
```

.. code-block:: bash

   ros2 run rclcpp_components component_container_mt --ros-args -p thread_num:=4

> * ``component_container_isolated`` with ``MultiThreadedExecutor`` for each component.

Reviewer: ```suggestion
* ``component_container_isolated`` with ``MultiThreadedExecutor`` for each component.

  * Use the ``--use_multi_threaded_executor`` argument to select ``MultiThreadedExecutor`` instead of ``SingleThreadedExecutor``.
```

Now that we have seen the basic operation of components, we can discuss a few more advanced topics.

> Component Container Types

Reviewer: This page doesn't seem to capitalize the words in subsection titles

```suggestion
Component container types
```
